src/helpers/git_helpers/branch_configuration_helpers.sh

```diff
@@ -94,89 +123,72 @@ function echo_update_child_branches {
 
 # Makes sure that we know all the parent branches
 # Asks the user if necessary
-# Aborts the script if not all branches become known.
+#
+# branches - space seperated list of branches
 function ensure_knows_parent_branches {
-  local current_branch=$1
-
-  if [ "$(knows_all_ancestor_branches "$current_branch")" = true ]; then
-    return
-  fi
-
-  # Here we don't have the ancestors list --> make sure we know all ancestors, then recompile it from all ancestors
+  local branches=$1
+  local branch
+  local child
```

Comment:
These are just for variables that get reassigned many times throughout the life of the function. That seems like a separate issue to me but I would be okay with it